src/Redirect/index.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * @adonisjs/http-server
+ *
+ * (c) Harminder Virk <virk@adonisjs.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+/// <reference path="../../adonis-typings/index.ts" />
+
+import { parse } from 'url'
+import encodeurl from 'encodeurl'
+import { IncomingMessage } from 'http'
+
+import { ResponseContract } from '@ioc:Adonis/Core/Response'
+import { RedirectContract } from '@ioc:Adonis/Core/Redirect'
+import { RouterContract, MakeUrlOptions } from '@ioc:Adonis/Core/Route'
+
+export class Redirect implements RedirectContract {
+  private forwardQueryString = false
+  private statusCode = 302
+  private queryString = new Map<string, any>()
+
+  constructor (
+    private request: IncomingMessage,
+    private response: ResponseContract,
+    private router: RouterContract
+  ) {
+  }
+
+  private computeQueryString (): string {
+    return Array.from(this.queryString.entries())
```

Comment:
We should use [qs.stringify](https://www.npmjs.com/package/qs), it generates a spec complaint query string